lms/static/js/learner_dashboard/views/course_entitlement_view.js

```diff
@@ -0,0 +1,395 @@
+                     Instantiates an instance of the Bootstrap v4 dialog modal and attaches it to the passed in element.
+
+                     This dialog acts as the second step in verifying the user's action to select, change or leave an
+                     available course session.
+                      */
+                     var confirmationMsgTitle,
+                         confirmationMsgBody,
+                         popoverDialogHtml,
+                         currentSessionId = this.entitlementModel.get('currentSessionId'),
+                         newSessionId = this.$('.session-select').find('option:selected').data('session_id');
+
+                     // Update the button popover text to enable two step authentication.
+                     if (newSessionId) {
+                         confirmationMsgTitle = !currentSessionId ?
+                             gettext('Are you sure that you would like to select this session?') :
+                             gettext('Are you sure that you would like to switch session?');
+                         confirmationMsgBody = !currentSessionId ? '' :
+                             gettext('Please know that by choosing to switch session you will' +
+                                 ' lose any progress you have made in this session.');
+                     } else {
+                         confirmationMsgTitle = gettext('Are you sure that you would like to leave this session?');
+                         confirmationMsgBody = gettext('Please know that by leaving you will lose any progress you ' +
+                                'had made in this session.');
+                     }
+
+                     // Remove existing popover and re-initialize
+                     popoverDialogHtml = this.verificationTpl({
+                         confirmationMsgTitle: confirmationMsgTitle,
+                         confirmationMsgBody: confirmationMsgBody
+                     });
+
+                     invokingElement.popover({
+                         placement: 'bottom',
+                         container: this.$el,
+                         html: true,
+                         trigger: 'click',
+                         content: popoverDialogHtml.text
+                     });
+                 },
+
+                 removeDialog: function(invokingElement) {
+                     /* Removes the Bootstrap v4 dialog modal from the update session enrollment button. */
+                     invokingElement.popover('dispose');
+                 },
+
+                 showDialog: function(invokingElement) {
+                     /* Given an element with an associated dialog modal, shows the modal. */
+                     invokingElement.popover('show');
+                 },
+
+                 hideDialog: function(invokingElement) {
+                     /* Hides the  modal without removing it from the DOM. */
+                     invokingElement.focus().popover('hide');
```

Comment:
Why do we call `focus` here, but not in `showDialog` or `removeDialog`?

Reply:
It was due to the binding of the 'this' object, the invoking element is the thing we want to focus on, but in the other case we want to focus on something outside of the invoking element. We could bind the function but it seemed more confusing to mess with the scope than just to set it elsewhere.